src/DependencyInjection/CKSourceCKFinderExtension.php

```diff
@@ -56,10 +58,31 @@ public function load(array $configs, ContainerBuilder $container): void
         $container->setParameter('ckfinder.connector.class', $config['connector']['connectorClass']);
         $container->setParameter('ckfinder.connector.auth.class', $config['connector']['authenticationClass']);
         $container->setParameter('ckfinder.connector.config', $config['connector']);
+
+        $this->registerServicesMap($config, $container);
     }
 
     public function getAlias(): string
     {
         return 'ckfinder';
     }
+
+    private function registerServicesMap(array $config, ContainerBuilder $container): void
+    {
+        $servicesMap = [];
+
+        foreach ($config['connector']['backends'] as $backend) {
+            if($backend['adapter'] === 's3') {
+                if (is_string($clientId = $backend['client'] ?? null)) {
+                    if (null === ($servicesMap[$clientId] ?? null)) {
+                        $servicesMap[$clientId] = new Reference($clientId);
+                    }
+                }
+            }
+        }
```

Comment:
Maybe there is a better way to detect services references... who knows

Reply:
![who knows](https://user-images.githubusercontent.com/2103975/175261114-5dbcc121-95ea-487d-a5e3-0051e203e2a6.gif)